src/main/java/bio/terra/janitor/db/JanitorDao.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.janitor.db;
+
+import bio.terra.janitor.app.configuration.JanitorJdbcConfiguration;
+import bio.terra.janitor.common.JanitorResourceTypeEnum;
+import bio.terra.janitor.common.exception.DuplicateLabelException;
+import bio.terra.janitor.common.exception.DuplicateTrackedResourceException;
+import java.time.OffsetDateTime;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+public class JanitorDao {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Autowired
+  public JanitorDao(JanitorJdbcConfiguration jdbcConfiguration) {
+    jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+  }
+
+  /** Creates the tracked_resource record and adding labels. */
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public UUID createResource(
+      String cloudResourceUid,
+      JanitorResourceTypeEnum resourceType,
```

Comment:
Like my previous comment, I feel the janitorDao shouldn't do any data process work, it will just purely do the db logic - "insert" operation here. And data process depends on the upstream method who calls it.  So all information need to be provided explicitly when it reaches to the db layer, service layer handles converting request to db format, validations.

Reply:
I disagree that CloudResourceUid serialization is not db specific logic. We logically want to store what's represented by the CloudResourceUid - we want to query based on that and get back answers that include CloudResourceUid. The JanitorDao is in charge of handling how to convert CloudResourceUid into a format that the database understands - which involves serializing it to a json string to use as jsonb postgres type.
Request parsing and validation are aspects of the service layer, so a `isValid(CloudResourceUid resource)` method is important to have for that layer, and the JanitorDao can state and assume it gets validated CloudResourceUids.

Reply:
ah, okay, moved those to Dao. And when I didn't see if there is a need to add isValid() now. The logic will just do the serizie again? And I added a static helper classes, exception would be thrown there.